src/dsp_tools/utils/xmlupload/write_diagnostic_info.py

```diff
@@ -3,117 +3,54 @@
-    save_location.mkdir(parents=True, exist_ok=True)
-    return save_location, server_as_foldername, timestamp_str
-
-
-def write_id2iri_mapping_and_metrics(
+def write_id2iri_mapping(
     id2iri_mapping: dict[str, str],
-    metrics: Optional[list[MetricRecord]],
-    failed_uploads: list[str],
-    input_file: Union[str, Path, etree._ElementTree[Any]],
+    input_file: str | Path | etree._ElementTree[Any],
     timestamp_str: str,
-    server_as_foldername: str,
-) -> bool:
-    """
-    Writes the id2iri mapping and the metrics into the current working directory,
-    and prints the failed uploads (if applicable).
-
-    Args:
-        id2iri_mapping: mapping of ids from the XML file to IRIs in DSP (initially empty, gets filled during the upload)
-        metrics: list with the metric records collected until now (gets filled during the upload)
-        failed_uploads: ids of resources that could not be uploaded (initially empty, gets filled during the upload)
-        input_file: path to the XML file or parsed ElementTree
-        timestamp_str: timestamp for the name of the log files
-        server_as_foldername: simplified version of the server URL that can be used as folder name
-
-    Returns:
-        True if there are no failed_uploads, False otherwise
-    """
-    # determine names of files
-    if isinstance(input_file, (str, Path)):
-        id2iri_filename = f"{Path(input_file).stem}_id2iri_mapping_{timestamp_str}.json"
-        metrics_filename = f"{timestamp_str}_metrics_{server_as_foldername}_{Path(input_file).stem}.csv"
-    else:
-        id2iri_filename = f"{timestamp_str}_id2iri_mapping.json"
-        metrics_filename = f"{timestamp_str}_metrics_{server_as_foldername}.csv"
-
-    # write files and print info
-    success = True
+) -> None:
+    """Writes the mapping of internal IDs to IRIs to a file."""
+    match input_file:
+        case str() | Path():
+            id2iri_filename = f"{Path(input_file).stem}_id2iri_mapping_{timestamp_str}.json"
+        case _:
+            id2iri_filename = f"{timestamp_str}_id2iri_mapping.json"
     with open(id2iri_filename, "x", encoding="utf-8") as f:
         json.dump(id2iri_mapping, f, ensure_ascii=False, indent=4)
         print(f"The mapping of internal IDs to IRIs was written to {id2iri_filename}")
         logger.info(f"The mapping of internal IDs to IRIs was written to {id2iri_filename}")
-    if failed_uploads:
-        print(f"\nWARNING: Could not upload the following resources: {failed_uploads}\n")
-        logger.warning(f"Could not upload the following resources: {failed_uploads}")
-        success = False
-    if metrics:
-        os.makedirs("metrics", exist_ok=True)
-        df = pd.DataFrame(metrics)
-        df.to_csv(f"metrics/{metrics_filename}")
-        print(f"Total time of xmlupload: {sum(int(record.duration_ms) for record in metrics) / 1000:.1f} seconds")
 
-    return success
+
+def write_metrics(
```

Comment:
@Nora-Olivia-Ammann @BalduinLandolt Not the topic of this PR, but just FYI: We once added some metrics to the xmlupload, which I guess are now redundant, because you are going to implement better measurements. The metrics we implemented back then were made "by hand", i.e. there are a lot of lines all over the code doing time measurements. If you want, you can remove them.

Reply:
good to know, probably makes sense to do that, but I will keep that for a separate task, I think